packages/typescript-client/src/offset.ts

```diff
@@ -0,0 +1,26 @@
+import { Offset } from './types'
+
+type ComparableOffset = [number, number]
+
+/**
+ * Compares two offsets and returns appropriate number for sorting
+ * comparisons
+ *
+ * @param offsetA
+ * @param offsetB
+ * @returns -1 if offsetA < offsetB, 1 if offsetA > offsetB, 0 if equal
+ */
+export function compareOffset(offsetA: Offset, offsetB: Offset): 1 | 0 | -1 {
+  const [oAx, oAy] = splitOffset(offsetA)
+  const [oBx, oBy] = splitOffset(offsetB)
+  if (oAx > oBx) return 1
+  if (oAx < oBx) return -1
+  if (oAy > oBy) return 1
+  if (oAy < oBy) return -1
+  return 0
+}
```

Comment:
Comparator functions in JS are not limited to returning 1, 0, or -1.
In general, they should return negative number if A < B, zero when equal, and a positive number when A > B.
This is very convenient as it allows comparison to be implemented as `A - B`.
For this example, that means we can simplify it to:

```suggestion
export function compareOffset(offsetA: Offset, offsetB: Offset): number {
  const [a1, a2] = splitOffset(offsetA)
  const [b1, b2] = splitOffset(offsetB)
  return a1 - b1 || a2 - b2
}
```

Reply:
whoops sorry this is no longer used anywhere so I'm going to remove the whole module